sys/include/checksum/crc32.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2022 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_checksum_crc32 CRC32
+ * @ingroup     sys_checksum
+ *
+ * @brief       CRC32 checksum algorithm implementation according to IEEE standards
+ * @{
+ *
+ * @file
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef CHECKSUM_CRC32_H
+#define CHECKSUM_CRC32_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   CRC-32 checksum
+ *
+ * Uses the `0xedb88320` polynomial
+ *
+ * @note enable the `crc32_fast` module for a look-up table based implementation.
```

Comment:
```suggestion
 * @note enable the `crc32_fast` module for a look-up table based implementation
         that trades code size for speed.
```

I have to admit that any capable programmer should be aware of the trade-off a look-up table provides, but maybe a few people profit from it being spelled out.